### ISSUE TITLE
Unwrap expliticly given selector before generating the name for the pipeline

### ISSUE THREAD

**Issue by kachayev** [scikit-adaptation/skada#28]
`make_da_pipeline` function right now gives an option to provide as a step estimator already wrapped into one of the selectors. When we use `_name_estimators`, it generates rather weird name. Better approach would be to unwrap estimator from the selector before naming it.

**Comment by kachayev** [comment]
Make sure #29 is merged before working on this issue.

**Comment by YanisLalou** [comment]
Can you give examples of "weird names"?

In this example:
```
{'otmappingadapter': Shared(base_estimator=OTMappingAdapter(), max_iter=100000, metric='sqeuclidean',
        norm=None),
 'perdomain-1': PerDomain(base_estimator=PCA(), copy=True, iterated_power='auto',
           n_components=None, n_oversamples=10,
           power_iteration_normalizer='auto', random_state=None,
           svd_solver='auto', tol=0.0, whiten=False),
 'perdomain-2': PerDomain(C=1.0, base_estimator=LogisticRegression(), class_weight=None,
           dual=False, fit_intercept=True, intercept_scaling=1, l1_ratio=None,
           max_iter=100, multi_class='auto', n_jobs=None, penalty='l2',
           random_state=None, solver='lbfgs', tol=0.0001, verbose=0,
           warm_start=False)}
           
```
           
Do we want to have instead of  `perdomain-1` and ` perdomain-2,` `pca` and `log_reg` respectively ?

**Comment by rflamary** [comment]
perdomain_pca maybe?

**Comment by kachayev** [comment]
Maybe we can do `perdomain_*` for `PerDomain` selector but remove `shared_*` for `Shared`? I guess, if you put `PerDomain` explicitly, you know what you are doing and what to expect. But `Shared` is added "automatically" (most likely), so you won't know how to refer to your estimator in the pipeline.

**Comment by kachayev** [comment]
Closed in #51
